app.py

```diff
@@ -1,23 +1,46 @@
 import json
 from flask import Flask, request, jsonify
 from flask.views import MethodView
-from werkzeug.routing import ValidationError
+from werkzeug.exceptions import HTTPException
+from HandledError import HandledError
 
 app = Flask(__name__)
+pattern = {'id': int, 'name': str, 'address': str, "work_time": str, "phone_number": str}
 with open("restaurants.json", "r") as read_file:
     data = json.load(read_file)
 
 restaurants = data['restaurants']
 
 
+def validate_post(content):
+    if content is None:
+        raise HandledError(status_code=400, description="Required data not available", name="Bad Request")
+    if content.keys() != pattern.keys():
```

Comment:
The order in which keys are returned is not fixed, so the keys should be compared as sets:
If set(...) != set(...)

Reply:
I decided that the order was not important. The main thing is that all keys are equal. For us, it doesn't matter in what order the data is written, does it?

Reply:
It does. 
"The main thing is that all keys are equal" - that is exactly set comparison case.
consider 2 lists: `['a', 'b', 'c']` and `['c', 'b', 'a']`. With `==` comparison you will have `False`, because order is different, but `set(['a', 'b', 'c']) == set(`['c', 'b', 'a'])` ==> `True`

Reply:
dict.keys() returns dict_keys object. It is behaves like set in this situation.

Reply:
![Снимок экрана (425)](https://user-images.githubusercontent.com/48601427/104566039-2d308080-565e-11eb-9831-15d11ea48a73.png)

Reply:
Wow, cool, I did not know that! Have read about it in the documentation, could you send a link?

Reply:
Sure:) https://docs.python.org/3/library/stdtypes.html#dictionary-view-objects

Reply:
Great, thanks! will know now :)